src/views/Permissions.vue

```diff
@@ -0,0 +1,296 @@
+import { UtilService } from '@/services/UtilService';
+import emitter from "@/event-bus";
+import { PermissionService } from '@/services/PermissionService';
+import { DateTime } from 'luxon';
+
+export default defineComponent({
+  name: 'Permissions',
+  components: {
+    IonButton,
+    IonButtons,
+    IonContent,
+    IonHeader,
+    IonIcon,
+    IonItem,
+    IonLabel,
+    IonList,
+    IonNote,
+    IonPage,
+    IonTitle,
+    IonToolbar,
+    PermissionItems
+  },
+  data() {
+    return {
+      securityGroupUsers: {} as any
+    }
+  },
+  computed: {
+    ...mapGetters({
+      securityGroups: 'util/getSecurityGroups',
+      permissionsByGroupType: 'permission/getPermissionsByGroupType',
+      currentGroupPermissions: 'permission/getCurrentGroupPermissions',
+      currentGroup: 'permission/getCurrentGroup',
+      allPermissions: 'permission/getAllPermissions'
+    })
+  },
+  async mounted() {
+    await this.store.dispatch('util/getSecurityGroups')
+    if(!this.allPermissions.length) await this.store.dispatch('permission/getAllPermissions')
+    if(!Object.keys(this.permissionsByGroupType).length) await this.store.dispatch('permission/getPermissionsByGroupType')
+    if(Object.keys(this.currentGroup).length) await this.store.dispatch('permission/getPermissionsByGroup', this.currentGroup.groupId)
+    await this.store.dispatch('permission/checkAssociated')
+  },
+  methods: {
+    createGroup() {
+      this.$router.replace({ path: `/create-security-group/` })
+    },
+    async updateCurrentGroup(group: any) {
+      emitter.emit('presentLoader')
+      await this.store.dispatch('permission/updateCurrentGroup', group)
+      await this.store.dispatch('permission/getPermissionsByGroup', this.currentGroup.groupId)
+      await this.store.dispatch('permission/checkAssociated')
+      await this.getUsersCount()
+      emitter.emit('dismissLoader')
+    },
+    async editGroupName() {
+      const alert = await alertController.create({
+        header: translate("Rename Security Group"),
+        inputs: [{
+          name: "groupName",
+          value: this.currentGroup.groupName
+        }],
+        buttons: [{
+          text: translate("Cancel"),
+          role: "cancel"
+        },
+        {
+          text: translate("Confirm"),
+          handler: async (data: any) => {
+            if (data.groupName) {
+              emitter.emit('presentLoader')
+
+              try {
+                const resp = await UtilService.updateSecurityGroup({
+                  groupId: this.currentGroup.groupId,
+                  groupName: data.groupName
+                })
+
+                if (!hasError(resp)) {
+                  showToast(translate("Security group renamed successfully."))
+                  await this.store.dispatch('util/getSecurityGroups')
+                  await this.store.dispatch('permission/updateCurrentGroup', this.securityGroups.find((group: any) => group.groupId === this.currentGroup.groupId))
```

Comment:
I think we should not make an api call just to get the updated name of the security group, we can simply update the name in the state instead of making this api calls.
Is there some other case for which the calls are made?